tendermint/spn
# test: Simulation and Makefile enhancements

Closes #367 

### What does this PR does?

Adds new `Makefile` targets for more simulation options.  Add a new non-determinism test which runs multiple sims to verify the state is the same multiple times.

Modifies the `simulation.yml` workflow so that it runs a `Makefile` target, but in effect the test is the same simulation with the same parameters.

### How to test?

Run the new targets:
- `make test-race`
- `make test-cover`
- `make bench`
- `make test-sim-nondeterminism`
- `make test-sim-ci`
- `make test-sim-benchmark`
- `make test-sim-profile`

Note that the larger sims (nondeterminism, benchmark, profile) run into the out of gas issue described in #369. This will need to be addressed in a new PR.
